### ISSUE TITLE
Rename dimension name before ingestion

### ISSUE THREAD

**Issue by meazza** [apache/incubator-druid#6281]
The dimensions of DimensionSpec can only configure name and type. 
If I have a json with field named countryId, and wish to rename the data stored in druid as country_id, it can not be satisfied.
So, is it possible to support renaming the field before ingestion? thanks~

**Comment by QiuMM** [comment]
You can rename the field at query time, like this:
```
{
  "type" : "default",
  "dimension" : <dimension>,
  "outputName": <output_name>,
  "outputType": <"STRING"|"LONG"|"FLOAT">
}
```

**Comment by meazza** [comment]
That's right, however, in cross team project, users need to change the column name of Druid data, to provide the exact schema as needed. I guess it may be a general demand.

**Comment by himanshug** [comment]
you are probably looking for http://druid.io/docs/latest/ingestion/transform-spec.html

**Comment by meazza** [comment]
@himanshug Thanks for your guide. I've tried this spec, and it does change the output name while ingestion.
In transformSpec, for digit column, I must specify an operator for it, like abs() or power(), how could I just keep the value of digit column and does not use any operator?

**Comment by milimetric** [comment]
Writing the name of the column seems to fail for us, do you have a working example?  We tried passing a string literal and that works, so the spec is fine except it doesn't seem to work to rename as @meazza asked for here.

**Comment by gianm** [comment]
Writing the name of the column as an expression should definitely work, but watch out for potential need to escape special characters.  You might need to wrap the column name in double quotes.

**Comment by milimetric** [comment]
hm, we do seem to have a problem here.  We tried `"expression": "plainName"`, `"expression": "\"nameInQuotes\""`, and `"expression": "'literal'"`.  The `'literal'` works, but the other two fail and we can't find anything in the logs.  How should we look for logs for this?  The documentation is not very clear on where errors would show up.  I'd like to post an error message to clarify this problem.

**Comment by milimetric** [comment]
Oh, I think this might be our issue, it's exactly what we're seeing: https://github.com/apache/incubator-druid/issues/7169

**Comment by gianm** [comment]
Ah, in that case, it should be fixed in 0.14.0 if you use a "parquet" parse spec.
